tests/test_downloader.py

```diff
@@ -403,3 +416,106 @@ async def parse(self, response, page: ItemPage):
         yield crawler.crawl()
 
     assert items == [{"a": "a"}]
+
+
+@inlineCallbacks
+def test_parse_callback_none(caplog) -> None:
+    """If request.callback == None, the DummyResponse annotation in the parse()
+    method would be ignored.
+
+    This means that the response is actually be downloaded as opposed to being
+    skipped.
+
+    See: https://github.com/scrapinghub/scrapy-poet/pull/106
+    """
+
+    collected = {}
+
+    with MockServer(EchoResource) as server:
+
+        class TestSpider(Spider):
+            name = "test_spider"
+            start_urls = [server.root_url]
+
+            custom_settings = {
+                "DOWNLOADER_MIDDLEWARES": {
+                    "scrapy_poet.InjectionMiddleware": 543,
+                },
+            }
+
+            def parse(self, response: DummyResponse):
+                collected["response"] = response
+
+        crawler = make_crawler(TestSpider, {})
+
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
can you use pytest.warns?

Reply:
https://github.com/scrapinghub/scrapy-poet/pull/106/commits/e353f81df552fea79d7ec12a394c9b6509bc2db2